python/lsst/daf/butler/datastores/chainedDatastore.py

```diff
@@ -1116,6 +1116,11 @@ def transfer_from(
             # child datastores.
             source_datastores = (source_datastore,)
 
+        refs = list(refs)
+        if len(refs) == 0:
```

Comment:
```python
if not refs:
```
does pass mypy for an Iterator but not sure if that's safe.

Reply:
If `refs` is a list then `not refs` and `len(refs) == 0` work identically.

Reply:
Yes. That's my point, but in the more general case for Iterator am I setting myself up for pain if I do that. I am wondering if we should make the type annotation be `Collection` instead and therefore guarantee that `len` works, or do something like:

```python
if not isinstance(refs, Sized):
    refs = list(refs)
```

Reply:
If we need `len` then `Collection` annotation is probably better.